tests/integration/helpers/utils.py

```diff
@@ -43,3 +48,19 @@ def _delete_record(pid_type, pid_value):
         object_uuid == PersistentIdentifier.object_uuid).delete()
 
     db.session.commit()
+
+
+@contextmanager
+def mock_addresses(addresses, mocked_local=False):
+    with requests_mock.Mocker() as requests_mocker:
+        if not mocked_local:
+            requests_mocker.register_uri(
+                requests_mock.ANY,
+                re.compile('.*(indexer|localhost).*'),
+                real_http=True,
+            )
+
+        for address in addresses:
+            requests_mocker.register_uri(**address)
+
+        yield
```

Comment:
I will also try to see if this (unrelated) refactor can be spun out in its own commit.

Reply:
Practicality beat purity and I left it in.